From d07ffbc777b405aeac98db7ea475ef697d5aee00 Mon Sep 17 00:00:00 2001
From: Michael Brown <mbrown@fensystems.co.uk>
Date: Sat, 16 May 2020 19:14:45 +0100
Subject: [PATCH 3/3] rpm: Store OAuth2 encryption key in a data file rather
 than in code

SuiteCRM generates an OAuth2 encryption key and stores it by modifying
the code file containing the ApiConfig class.  This is unbelievably
dumb.

Question to SuiteCRM developers: when you find yourself writing code
that reads in another PHP code file, performs a string substitution,
and writes it back out, do you not ever think "there must be a better
way to do this"?  Like, for example, using an actual configuration
file.  That way, you can upgrade the code without destroying your
configuration, or vice versa.  It's magic!

It's also really not that hard.  The rest of the world has been doing
it the right way since at least the 1960s, so you've only got half a
century of coding practices to catch up on.  Get here when you can.

Signed-off-by: Michael Brown <mbrown@fensystems.co.uk>
---
 Api/Core/Config/ApiConfig.php | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Api/Core/Config/ApiConfig.php b/Api/Core/Config/ApiConfig.php
index d6db2a003..539639e25 100644
--- a/Api/Core/Config/ApiConfig.php
+++ b/Api/Core/Config/ApiConfig.php
@@ -1,6 +1,8 @@
 <?php
 namespace Api\Core\Config;
 
+define('OAUTH2_ENCRYPTION_KEY_VALUE', file_get_contents('/etc/suitecrm/api.key');
+
 class ApiConfig
 {
     // we still support 5.5.9
@@ -18,7 +20,7 @@ class ApiConfig
 
     const OAUTH2_PRIVATE_KEY = 'Api/V8/OAuth2/private.key';
     const OAUTH2_PUBLIC_KEY = 'Api/V8/OAuth2/public.key';
-    const OAUTH2_ENCRYPTION_KEY = '';
+    const OAUTH2_ENCRYPTION_KEY = OAUTH2_ENCRYPTION_KEY_VALUE;
     
     /**
      *
-- 
2.25.1

